funcgen-cd/funcgen.go

```diff
@@ -0,0 +1,500 @@
+		"go mod vendor",
+		"go mod edit -dropreplace="+repo,
+		// test that deps are treated correctly
+		"go build -mod=vendor",
+	); err != nil {
+		return err
+	}
+
+	// Google: Warning: If you have both a go.mod file and a vendor directory at the root of your project,
+	// the contents of the vendor directory will be ignored when your function is built in the cloud.
+	// To ensure that your vendor directory is used,
+	// you must exclude the go.mod file from your project’s source code prior to deployment.
+	if err := os.Remove(filepath.Join(outDirPath, "go.mod")); err != nil {
+		return err
+	}
+	if err := os.Remove(filepath.Join(outDirPath, "go.sum")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func execCmdsInDir(dir string, commands ...string) error {
+	for _, c := range commands {
+		if err := execCmd("cd " + dir + " ; " + c); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// execCmd wraps exec.Command execution
+func execCmd(command string) error {
+	cmd := exec.Command("bash", "-c", command)
+
+	data, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("err: %v, details: %s", err, string(data))
+	}
+	return nil
+}
+
+func zipDir(baseFolder, outPath string) error {
+	outFile, err := os.Create(outPath)
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	w := zip.NewWriter(outFile)
+	if err := addFiles(w, baseFolder, ""); err != nil {
+		return err
+	}
+	return w.Close()
+}
+
+func addFiles(w *zip.Writer, basePath, baseInZip string) error {
+	files, err := ioutil.ReadDir(basePath)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		if strings.Contains(file.Name(), ".git") {
+			continue
+		}
+		filePath := filepath.Join(basePath, file.Name())
+		if !file.IsDir() {
+			dat, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+
+			f, err := w.Create(filepath.Join(baseInZip, file.Name()))
+			if err != nil {
+				return err
+			}
+			_, err = f.Write(dat)
+			if err != nil {
+				return err
+			}
+		} else if file.IsDir() {
+			if err := addFiles(w, filePath, filepath.Join(baseInZip, file.Name())); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func hashRenameZipFile(path string) (string, error) {
+	hash, err := hashFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	dir, base := filepath.Dir(path), filepath.Base(path)
+	base = strings.Replace(base, ".zip", "."+hash+".zip", -1)
+
+	newPath := filepath.Join(dir, base)
+	if err := os.Rename(path, newPath); err != nil {
+		return "", err
+	}
+	return newPath, nil
+}
+
+func hashFile(path string) (string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hash := md5.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return "", err
+	}
+	hashInBytes := hash.Sum(nil)[:16]
+	return hex.EncodeToString(hashInBytes), nil
+}
+
+func exportVarsFile(m map[string]string, path string) error {
+	s := &terraformVar{Sources: m}
```

Comment:
```suggestion
	s := struct{
		Sources map[string]string `json:"func_sources_zip"`
	}{m}
```
And remove `terraformVar` type (it's used only here).

Reply:
Done.